src/Models/Constants/Types.ts

```diff
@@ -249,6 +250,7 @@ export type AzureResourceSearchParams = {
         storageAccountBlobUrl?: string;
         [key: string]: any;
     };
+    isAdditionalSearchParamsRequired?: boolean;
```

Comment:
it is the `additionalParams` one above in that typing definition. Instead I can make two types like WithAdditionalParams | WithoutAdditionalParams to ensure dependency in parameters?

Reply:
that's always my personal preference since it makes it more obvious to someone consuming which args are dependent on eachother. It's kind of self-documenting and stupid proofs someone from using it in a way you didn't intend by mistake.

Reply:
added discriminating union types here too